### PR TITLE
add additional keywords and change scope of symbol names

### PR DESCRIPTION
**Pull request by tcpluess**

### CODE REVIEW

File: LinkerScript.sublime-syntax [hunk 1]
```diff
@@ -5,10 +5,12 @@ file_extensions: [ld]
 scope: source.linker
 
 variables:
-  num_suffix: '[KM]'
+  num_suffix: '[kKM]'
```
Comment (jbw3) [inline LinkerScript.sublime-syntax:8]: If we're going to support lower case `k` then I think we should add lower case `m` as well.

File: LinkerScript.sublime-syntax [hunk 2]
```diff
@@ -72,7 +76,7 @@ contexts:
               |
                 \B\.[A-Za-z_\.0-9\-]+
               )'
-      scope: variable.other.linker
+      scope: entity.name
```
Comment (jbw3) [inline LinkerScript.sublime-syntax:79]: According to Sublime Text's [scope naming documentation](https://www.sublimetext.com/docs/3/scope_naming.html), the `entity.` scope is more for types, structs, etc., so I think the `variable.` scope is better here.

File: LinkerScript.sublime-syntax [hunk 3]
```diff
@@ -82,6 +86,17 @@ contexts:
           scope: invalid.illegal.linker
           pop: true
 
+  strings:
```
Comment (jbw3) [inline LinkerScript.sublime-syntax:89]: In the GNU linker syntax, double quotes are for surrounding symbol names, so this `strings` section conflicts with the above `symbol_names` section.

File: LinkerScript.sublime-syntax [hunk 4]
```diff
@@ -82,6 +86,17 @@ contexts:
...
+    - match: '"'
+      push:
+        - meta_scope: string.quoted.double.linker
+        - match: '"'
+          pop: true
+
+  memory_modifiers:
+  - match: \([rwxail!]*[RWXAIL!]*\)
+    scope: storage.type
```
Comment (jbw3) [inline LinkerScript.sublime-syntax:98]: This scope needs to have `.linker` on the end.

File: LinkerScript.sublime-syntax (hunk 1 above)
Comment (tcpluess) [reply LinkerScript.sublime-syntax:8]: not really, as uppercase `K` is anyways wrong as SI prefix, whereas uppercase `M` is absolutely correct.

**Pull request closed without merging**